pep-0387.txt

```diff
@@ -112,6 +112,28 @@ Basic policy for backwards compatibility
   platforms).
 
 
+Soft Deprecation
+================
+
+A soft deprecation can be used when using an API must no longer be used
+to write new code, but it remains safe to continue using it in existing
+code. The API remains documented and tested, but will not be developed
+further (no enhancement).
+
+The documentation of a soft deprecation should explain why the API
+should be avoided, and if possible propose a replacement.
+
+Soft deprecated functions must emit a SoftDeprecationWarning warning
+(new in Python 3.13) in the `Python Development Mode
+<https://docs.python.org/dev/library/devmode.html>`_ and if `Python is
+built in debug mode
+<https://docs.python.org/dev/using/configure.html#python-debug-build>`_:
+by default, this warning is not emitted.
+
+Changing a soft deprecation into a regular "hard" deprecation must
+follow the regular `Backwards Compatibility Rules`_.
```

Comment:
Furthermore, a soft deprecation carries no implication that it will *ever* be turned into a hard deprecation. It's entirely acceptable for a soft deprecation to remain that way forever.

Reply:
I added a section to be more explicit that soft deprecation does NOT imply scheduling the removal of the API.

I wrote that converting a soft deprecation to a hard deprecation requires to follow the usual process to deprecate an API. Don't you think that it's enough to mean that a soft deprecated API can stay like that forever. If you don't that it's not enough, can you propose to rephase this paragraph?

Reply:
How about this?

> If the decision is made to deprecate (in the regular sense) a feature that is currently soft deprecated, the deprecation must follow the regular backward compatibility rules (i.e., there is no exception because the feature is already soft deprecated).